.github/workflows/push_tests.yml

```diff
@@ -0,0 +1,45 @@
+name: Run all tests
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  HF_HOME: /mnt/cache
+  OMP_NUM_THREADS: 8
+  MKL_NUM_THREADS: 8
+  PYTEST_TIMEOUT: 1000
+  RUN_SLOW: yes
+
+jobs:
+  run_tests_single_gpu:
+    name: Diffusers tests
+    strategy:
+      fail-fast: false
+      matrix:
+        machine_type: [ single-gpu ]
+    runs-on: [ self-hosted, docker-gpu, '${{ matrix.machine_type }}' ]
+    container:
+      image: nvcr.io/nvidia/pytorch:22.07-py3
+      options: --gpus 0 --shm-size "16gb" --ipc host -v /mnt/cache/.cache/huggingface:/mnt/cache/
+
+    steps:
+    - name: Checkout diffusers
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+
+    - name: NVIDIA-SMI
+      run: |
+        nvidia-smi
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install -e .[quality,test]
+        python -m pip install scipy transformers
+
+    - name: Run all (incl. slow) tests on GPU
+      run: |
+        python -m pytest -n 2 --max-worker-restart=0 --dist=loadfile -s
```

Comment:
In `transformers`, we specify what to test, like `tests` or `tests/models`. Here I don't see the target. Does this work? Even so, maybe specify it would be better.

Reply:
A bit worried about `-n 2` for running tests `GPU`. But we can just keep this in mind.